prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -440,14 +440,18 @@ class Hl7Serializer(
         setLiterals(terser)
         // serialize the rest of the elements
         report.schema.elements.forEach { element ->
-            val value = report.getString(row, element.name).let {
+            var value = report.getString(row, element.name).let {
                 if (it.isNullOrEmpty()) {
                     element.default ?: ""
                 } else {
                     it
                 }
             }
 
+            if (element.hl7Field in ST_FIELDS_UNIVERSAL_CHAR_LIMIT_50) {
```

Comment:
This doesn't feel like the right place to do this truncation because it misses the hl7OutputFields cases. My suggestion would be to:
1. Create a separate function to encapsulate the truncation logic. eg `trunacateValue(..)`
2. Make this new function be a place for all truncation logic for `ST_FIELDS_UNIVERSAL_CHAR_LIMIT_50` and `HD_FIELDS_LOCAL`.
3. Call this new function in both hl7Field and hl7OutputFields cases.

Reply:
Thanks, will do that. I'm trying to see where it affects the property for truncating the HD fields. But I will encapsulate the truncation process.

Reply:
I will work on the hl7OutputFields part.